utils/preferences/tests/dtf_tests.rs

```diff
@@ -0,0 +1,70 @@
+mod dtf;
+
+use icu_datetime::options::length;
+use icu_datetime::options::preferences::HourCycle;
+use icu_locid::{locale, subtags_language, subtags_region, Locale};
+
+use dtf::*;
+
+#[test]
+fn dtf_default() {
+    let loc = locale!("en-US");
+
+    let options = DateTimeFormatOptions {
+        time_length: Some(length::Time::Long),
+        ..Default::default()
+    };
+
+    let dtf = DateTimeFormat::new(loc.into(), options);
+
+    assert_eq!(dtf.format(0), String::from("12:13 am"));
+}
+
+#[test]
+fn dtf_uext() {
+    let loc: Locale = "en-US-u-hc-h11".parse().unwrap();
+    let dtf = DateTimeFormat::new(loc.into(), Default::default());
+    assert_eq!(
+        dtf.format(0),
+        String::from("Monday, June 23rd 2022, 00:13 am")
+    );
+}
+
+#[test]
+fn dtf_prefs() {
+    let loc: Locale = "en-US-u-hc-h11".parse().unwrap();
+    let mut prefs = DateTimeFormatPreferences {
```

Comment:
Name bikeshed:

- DateTimeFormatPreferences
- DateTimePreferences (based on crate/component name)
- DateTimeFormatterPrefs
- DateTimeLocale

Discussion:

- @zbraniecki - I don't know if I want to take the union of all formatters in a component into a preference bag. ZonedDateTimeFormatter needs more preferences than TimeFormatter, but then TimeFormatter takes on all those extra preferences.
- @echeran - There are benefits to having them all together. If you have prefs for scientific and currency in the same global preference bag, it's just that the currency formatter won't look at the scientific prefs.
- @zbraniecki - If we make crate-scoped preferences, then we're making a universe where the crate you live in determines how much extraneous information you get in your preference bag.